sdk/documentintelligence/ai-document-intelligence-rest/src/clientDefinitions.ts

```diff
@@ -0,0 +1,290 @@
+  ClassifyDocumentFromStreamDefaultResponse,
+  ClassifyDocument202Response,
+  ClassifyDocumentDefaultResponse,
+  GetClassifyResult200Response,
+  GetClassifyResultDefaultResponse,
+} from "./responses";
+import { Client, StreamableMethod } from "@azure-rest/core-client";
+
+export interface ListOperations {
+  /** Lists all operations. */
+  get(
+    options?: ListOperationsParameters
+  ): StreamableMethod<ListOperations200Response | ListOperationsDefaultResponse>;
+}
+
+export interface GetDocumentModelBuildOperation {
+  /** Gets operation info. */
+  get(
+    options?: GetDocumentModelBuildOperationParameters
+  ): StreamableMethod<
+    GetDocumentModelBuildOperation200Response | GetDocumentModelBuildOperationDefaultResponse
+  >;
+  /** Gets operation info. */
+  get(
+    options?: GetDocumentModelComposeOperationParameters
+  ): StreamableMethod<
+    GetDocumentModelComposeOperation200Response | GetDocumentModelComposeOperationDefaultResponse
+  >;
+  /** Gets operation info. */
+  get(
+    options?: GetDocumentModelCopyToOperationParameters
+  ): StreamableMethod<
+    GetDocumentModelCopyToOperation200Response | GetDocumentModelCopyToOperationDefaultResponse
+  >;
+  /** Gets operation info. */
+  get(
+    options?: GetDocumentClassifierBuildOperationParameters
+  ): StreamableMethod<
+    | GetDocumentClassifierBuildOperation200Response
+    | GetDocumentClassifierBuildOperationDefaultResponse
+  >;
+  /** Gets operation info. */
+  get(
+    options?: GetOperationParameters
+  ): StreamableMethod<GetOperation200Response | GetOperationDefaultResponse>;
+}
+
+export interface GetResourceInfo {
+  /** Return information about the current resource. */
+  get(
+    options?: GetResourceInfoParameters
+  ): StreamableMethod<GetResourceInfo200Response | GetResourceInfoDefaultResponse>;
+}
+
+export interface GetAnalyzeResult {
+  /** Gets the result of document analysis. */
+  get(
+    options?: GetAnalyzeResultParameters
+  ): StreamableMethod<GetAnalyzeResult200Response | GetAnalyzeResultDefaultResponse>;
+}
+
+export interface AnalyzeDocumentFromStream {
+  /** Analyzes document with document model. */
+  post(
+    options: AnalyzeDocumentFromStreamParameters
+  ): StreamableMethod<
+    AnalyzeDocumentFromStream202Response | AnalyzeDocumentFromStreamDefaultResponse
+  >;
+  /** Analyzes document with document model. */
+  post(
+    options: AnalyzeDocumentParameters
+  ): StreamableMethod<AnalyzeDocument202Response | AnalyzeDocumentDefaultResponse>;
+}
+
+export interface GetModel {
+  /** Gets detailed document model information. */
+  get(
+    options?: GetModelParameters
+  ): StreamableMethod<GetModel200Response | GetModelDefaultResponse>;
+  /** Deletes document model. */
+  delete(
+    options?: DeleteModelParameters
+  ): StreamableMethod<DeleteModel204Response | DeleteModelDefaultResponse>;
+}
+
+export interface BuildModel {
+  /** Builds a custom document analysis model. */
+  post(
```

Comment:
Could you add a case for this method? I notice the method `buildModel` is an LRO with `resourceLocation` and I am worrying about if we could support this case correctly(see [discussion](https://github.com/Azure/azure-rest-api-specs-pr/pull/15526#discussion_r1392990030)). 

tsp: https://github.com/Azure/azure-rest-api-specs-pr/blob/848f569574785ce962c31e7de25082e942b8486a/specification/cognitiveservices/DocumentIntelligence/routes.tsp#L179-L192

Reply:
this was working as expected though when I tested